onnxruntime/contrib_ops/cpu/bert/attention_cpu_base.h

```diff
@@ -20,27 +20,48 @@ class AttentionCPUBase : public AttentionBase {
 
   template <typename T>
   Status ApplyAttention(const T* Q,                           // Q data with shape BxNxSxH
-                        const T* K,                           // K data with shape BxNxSxH
-                        const T* V,                           // V value with size BxNxSxH_v
+                        const T* K,                           // K data with shape BxNxLxH
+                        const T* V,                           // V value with size BxNxLxH_v
                         const Tensor* mask_index,             // mask index. nullptr if no mask or its size is B
                         const Tensor* past,                   // past state
                         Tensor* output,                       // output tensor
+                        Tensor* present_k,                    // present K output tensor (if separating present KV)
+                        Tensor* present_v,                    // present V output tensor (if separating present KV)
                         int batch_size,                       // batch size (B)
-                        int sequence_length,                  // sequence length (S)
+                        int sequence_length,                  // sequence length of Q (S)
+                        int kv_sequence_length,               // sequence length of K or V (L)
                         int qk_head_size,                     // head size of Q or K (H)
                         int v_head_size,                      // head size of V (H_v)
                         int v_hidden_size,                    // hidden size of V (D_v)
                         const Tensor* relative_position_bias, // bias addition in QK. Its size is BxNxSxT
                         OpKernelContext* context) const {
-    const int kv_sequence_length = sequence_length;
 
     AllocatorPtr allocator;
     ORT_RETURN_IF_ERROR(context->GetTempSpaceAllocator(&allocator));
 
     auto* tp = context->GetOperatorThreadPool();
 
     int past_sequence_length = 0;
-    Tensor* present = GetPresent(context, past, batch_size, v_head_size, sequence_length, past_sequence_length);
+    OrtValue present_kv;
+    Tensor* present = nullptr;
+    bool separate_present_kv = present_k != nullptr && present_v != nullptr;
+    if (separate_present_kv) {
+      auto element_type = DataTypeImpl::GetType<T>();
+      past_sequence_length = (nullptr != past) ? static_cast<int>(past->Shape().GetDims()[3]) : 0;
```

Comment:
Instead of concatenating past_key and past_value, we shall change ComputeAttentionProbs to accept past_key and past_value.

Reply:
`ComputeAttentionProbs` only needs `present_key` because `ComputeVxAttentionScore` will need `present_value`